build.py

```diff
@@ -214,13 +295,41 @@ def _validate_android_args(args: argparse.Namespace):
                 )
 
         if args.cmake_generator != original_cmake_generator:
-            log.info(f"Setting CMake generator to '{args.cmake_generator}' for cross-compiling for Android.")
+            log.info(
+                f"Setting CMake generator to '{args.cmake_generator}' for cross-compiling for Android."
+            )
 
         # no C# on Android so automatically skip
         if not args.skip_csharp:
             args.skip_csharp = True
 
 
+def _validate_ios_args(args: argparse.Namespace):
+    if args.ios:
+        # TODO: cross compile for iOS on Linux?
+        if util.is_mac():
+            needed_args = [
+                args.ios_sysroot,
+                args.osx_arch,
+                args.apple_deployment_target,
```

Comment:
maybe think about the naming of these options - using "ios", "osx", and "apple" here seems inconsistent. if we want to make it general, maybe use "apple". if it's iOS-specific, use "ios".

Reply:
make sense - made them all "ios"